servlet-engine/src/main/java/io/micronaut/servlet/engine/DefaultServletHttpRequest.java

```diff
@@ -323,7 +337,18 @@ public String getContextPath() {
 
     @Override
     public InputStream getInputStream() throws IOException {
-        return servletByteBuffer != null ? servletByteBuffer.toInputStream() : delegate.getInputStream();
+        CompletableFuture<? extends CloseableAvailableByteBody> buffered = byteBody.buffered.get();
+        if (buffered == null) {
```

Comment:
when is this `null`?

Reply:
it's null when nobody called `.buffer()`, i.e. the normal case.

In the future I want to replace this with `byteBody.toInputStream()`, but that needs more impl work.